### PR TITLE
connection pooling

### PR DESCRIPTION
**Pull request by SamHausmann**

### CODE REVIEW

File: api/src/main/java/com/basistech/rosette/api/RosetteAPI.java [hunk 1]
```diff
@@ -1770,6 +1759,8 @@ private String headerValueOrNull(Header header) {
             throws IOException, RosetteAPIException {
         int status = httpResponse.getStatusLine().getStatusCode();
         String encoding = headerValueOrNull(httpResponse.getFirstHeader(HttpHeaders.CONTENT_ENCODING));
+        int concurrencyHeader = Integer.parseInt(headerValueOrNull(httpResponse.getFirstHeader("X-RosetteApi-Concurrency")));
```
Comment (guruxu) [inline api/src/main/java/com/basistech/rosette/api/RosetteAPI.java:1762]: You need to deal with `NumberFormatException` here which is unchecked.
Add some unit tests for no header, empty header, non-sense string, large number, negative number, etc to make sure things work ...
